### ISSUE TITLE
[openimageio] FindOpenImageIO.cmake does not separate Release/Debug configurations

### ISSUE THREAD

**Issue by janholt** [microsoft/vcpkg#8416]
Building sample program in Release configuration links with debug openimageio library.

**Environment**
Windows10, MS VS 2017 15.9.16

**To Reproduce**
```
vcpkg.exe install openimageio:x64-windows
```
Configure, Generate sample(empty `int main{}`) program with CMakeLists.txt:
```
cmake_minimum_required(VERSION 3.12)
project(test_oiio)
find_package(openimageio REQUIRED)
add_executable(test_oiio test_oiio.cpp)
target_include_directories(test_oiio PRIVATE ${OPENIMAGEIO_INCLUDE_DIR})
target_link_libraries(test_oiio ${OPENIMAGEIO_LIBRARIES})
```
Gives `test_oiio.vcxproj` with:
```
  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
    <Link>
      <AdditionalDependencies>D:\ws\vcpkg\probe\openimageio\vcpkg\installed\x64-windows\debug\lib\OpenImageIO.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;comdlg32.lib;advapi32.lib</AdditionalDependencies>
  ...
```

**Cause**
`find_package(openimageio)` uses `FindOpenImageIO.cmake` (through vcpkg-cmake-wrapper.cmake), which does `find_library ( OPENIMAGEIO_LIBRARY ...)` without selecting configuration.
There is openexr port, which also uses `FindOpenEXR.cmake`, but separates configurations by
```
include(SelectLibraryConfigurations)
if(NOT OpenEXR_BASE_LIBRARY)
  find_library(OpenEXR_BASE_LIBRARY_RELEASE NAMES IlmImf-${OpenEXR_LIB_SUFFIX})
  find_library(OpenEXR_BASE_LIBRARY_DEBUG NAMES IlmImf-${OpenEXR_LIB_SUFFIX}_d)
  select_library_configurations(OpenEXR_BASE)
endif()
```
`SelectLibraryConfigurations` solution has a drawback, that it does not propagate dependencies (notable with openimageio static build #8049 ). Is it better to get installed openimageio targets, and enable traditional `OPENIMAGEIO_LIBRARIES` on top of targets?

**Comment by janholt** [comment]
There is open PR in OpenImageIO homepage (https://github.com/OpenImageIO/oiio/pull/2352) with cmake refactoring (including mark target's link_libraries and include_directories with PUBLIC/PRIVATE).
It would be easy then to add install targets.

**Comment by JackBoosY** [comment]
Hi @janholt, thanks for reporting this issue!
I've fixed it in #8543, please update vcpkg and rebuild openimageio.

Thanks.
